src/arviz_stats/base/dataarray.py

```diff
@@ -193,5 +193,30 @@ def kde(self, da, dims=None, circular=False, grid_len=512, **kwargs):
         out = concat((grid, pdf), dim=plot_axis)
         return out.assign_coords({"bw" if da.name is None else f"bw_{da.name}": bw})
 
+    def thin(self, da, factor="auto", dims=None):
+        """Compute eti on DataArray input."""
+        if dims is None:
+            dims = rcParams["data.sample_dims"]
+
+        n_samples = da.sizes["chain"] * da.sizes["draw"]
+
+        if factor == "auto":
+            ess_ave = np.minimum(self.ess(da, method="bulk"), self.ess(da, method="tail")).mean()
+            factor = int(np.ceil(n_samples / ess_ave))
```

Comment:
```suggestion
        if factor == "auto":
            n_samples = np.product(da.sizes[dim] for dim in sample_dims)
            ess_ave = np.minimum(self.ess(da, method="bulk", dims=sample_dims), self.ess(da, method="tail", dims=sample_dims)).mean()
            factor = int(np.ceil(n_samples / ess_ave))
```

I would use the same logic here as in ess/mcse/rhat which basically boils down to `sample_dims` must be length 2, first one is "chain" the 2nd is "draw". And now that I write it we'll have to fix it, only rhat should do this, ess and mcse should work on 1d arrays/single dimension too.

A potential fix could be to do something closer to the array backend, where there is a `chain_axis` and a `draw_axis` but using `_dim` instead.

By default both are none, then:
  - sample_dims is a single dimension: chain_dim continues to be none, draw_dim=sample_dims
  - sample_dims is length 2: chain_dim=sample_dims[0], draw_dim=sample_dims[1]
  - sample_dims is length>2: error

Thinning then happens along `draw_dim` (modifying comment below).